imgcat/imgcat.go

```diff
@@ -85,37 +86,65 @@ func Inline(b bool) Option {
 	return Option(fmt.Sprintf("inline=%d", boolToInt(b)))
 }
 
-// New returns a writer that encodes images and for iterm2.
-func New(w io.Writer, options ...Option) io.WriteCloser {
+// IsSupported check whether imgcat works in the current terminal.
+func IsSupported() bool {
+	return os.Getenv("TERM_PROGRAM") == "iTerm.app"
+}
+
+// New returns a writer that encodes images for iterm2.
+func New(out io.Writer, options ...Option) (io.WriteCloser, error) {
```

Comment:
What about s/out/w? out sounds like a good internal name but w is fine for the public API.

Reply:
done